quantumblacklabs/kedro-viz
# [KED-2997] Create rename run and edit notes UI behavior

## Description

Resolves [KED-2997](https://jira.quantumblack.com/browse/KED-2997).

## Development notes

To do this, I've created a couple of new components:

Front-end : 

- A `<RunDetailsModal/>` component, based on the `<SettingsModal/>` that we already had, except this new one isn't connected in any way to Redux. It does share a lot of styles with the `<SettingsModal/>`, as I'm importing that CSS file.
- A new `<Input/>` component that lives within a new `components/ui` folder. This component actually uses a `<textarea/>` HTML element under the hood, which gives good flexibility. And further, going forward, I think we can start to move some UI-only components into this folder and create any new UI components here, as needed.
- A `<Pencil/>` icon component.

Backend : 
UpdateRunDetailSuccess now returns a Run with the latest changes requested by the user instead of returning a JSONObject as it did previously. 

## QA notes

Check out the branch and click on the Edit details pencil icon or the run metadata title/notes text to trigger the modal.

## Checklist
